src/moderation/services/moderatorActions/moderatorActions.service.ts

```diff
@@ -26,8 +25,11 @@ export class ModeratorActionsService implements IModeratorActionsService {
 
     public async banUser(payload: ModerationPayloadDto): Promise<void> {
         const user = await this._dbContext.Users.findUserById(payload.id);
-        if (user.roles.some(role => role === Role.ADMIN || role === Role.MODERATOR)) {

```

Comment:
According to your logic, if user (who's gonna get banned) was a moderator AND the authenticated user was an admin, your if statement will still rejects admin. 

(Even though in the auth guard we permit the user if they have an admin role, this banUser method is passed that guard already).